[UI v2] feat: Adds data and UX for task run concurrency limit

devinvillarosa opened the pull request:
### Checklist
1. Adds react queries for task-run-concurrency-limit
2. Adds UX for the task run tab. (Most UX was copied / pasted from global-concurrency-limit)

Will write tests for the page once I figure out how to get it to render correctly in rtl


https://github.com/user-attachments/assets/7b420859-1a26-45d8-a538-d367b2566147



- [x] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [x] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [x] If this pull request adds functions or classes, it includes helpful docstrings.


Relates to https://github.com/PrefectHQ/prefect/issues/15512

devinvillarosa commented on ui-v2/src/hooks/global-concurrency-limits.ts:
```diff
@@ -55,13 +54,6 @@ export const useListGlobalConcurrencyLimits = (
 	filter: GlobalConcurrencyLimitsFilter = { offset: 0 },
 ) => useSuspenseQuery(buildListGlobalConcurrencyLimitsQuery(filter));
 
-useListGlobalConcurrencyLimits.loader = ({
-	context,
```
Removing since this isn't used anymore in the ui

desertaxle commented on ui-v2/src/hooks/task-run-concurrency-limits.ts:
```diff
@@ -0,0 +1,205 @@
+
+// ----- 🔑 Queries 🗄️
+// ----------------------------
+export const buildListTaskRunConcurrencyLimitsQuery = (
+	filter: TaskRunConcurrencyLimitsFilter = { offset: 0 },
+) =>
+	queryOptions({
+		queryKey: queryKeyFactory.list(filter),
+		queryFn: async () => {
+			const res = await getQueryService().POST("/concurrency_limits/filter", {
+				body: filter,
+			});
+			return res.data ?? [];
+		},
+	});
```
We probably want some polling on this query to ensure the user sees relatively up-to-date info on how many slots are occupied while staying on this page. Somewhere between 30 and 60 seconds seems reasonable to me.

devinvillarosa replied:
all, let me update that

devinvillarosa commented:
Added a few more features:

1. Pagination in both tables that's done on the client side.
2. Added search via search params.
3. Added client-side filtering based on search params

desertaxle commented on ui-v2/src/components/concurrency/global-concurrency-view/data-table/data-table.tsx:
```diff
@@ -53,11 +59,36 @@ export const GlobalConcurrencyDataTable = ({
 	onEditRow,
 	onDeleteRow,
 }: Props) => {
+	const navigate = routeApi.useNavigate();
+	const { search } = routeApi.useSearch();
```
I think including a dependency on the router might make it harder to test this component, but I'm not sure, and we can refactor if that turns out to be the case.

desertaxle commented on ui-v2/src/components/concurrency/task-run-concurrenct-view/header.tsx:
```diff
@@ -0,0 +1,22 @@
+	return (
+		<div className="flex gap-2 items-center">
+			<Typography variant="h4">Task Run Concurrency Limits</Typography>
+			<Button
+				onClick={onAdd}
+				size="icon"
+				aria-label="add task run concurrency limit"
+			>
+				<Icon id="Plus" />
+			</Button>
```
I think we should make this an outline button and smaller to match the current style.
```suggestion
			<Button
				onClick={onAdd}
				size="icon"
				className="h-7 w-7"
				aria-label="add task run concurrency limit"
				variant="outline"
			>
				<Icon id="Plus" className="h-4 w-4" />
			</Button>
```

desertaxle commented on ui-v2/src/components/concurrency/task-run-concurrenct-view/dialogs/delete-dialog.tsx:
```diff
@@ -0,0 +1,69 @@
+
+	const handleOnClick = (id: string | undefined) => {
+		if (!id) {
+			throw new Error("'id' field expected in GlobalConcurrencyLimit");
```
nit, but we won't need this much longer.
```suggestion
			throw new Error("'id' field expected in TaskRunConcurrencyLimit");
```

desertaxle reviewed (approved):
A couple small comments, but otherwise LGTM!
